torchattacks/attacks/fmn.py

```diff
@@ -0,0 +1,338 @@
+import math
+from functools import partial
+from typing import Optional
+
+import torch
+import torch.nn as nn
+from torch import Tensor
+from torch.optim import SGD
+from torch.optim.lr_scheduler import CosineAnnealingLR
+
+from ..attack import Attack
+
+
+class FMN(Attack):
+    r"""
+    FMN in the paper 'Fast Minimum-norm Adversarial Attacks through Adaptive Norm Constraints'
+    [https://arxiv.org/abs/2102.12827]
+
+    Distance Measure : L0, L1, L2, Linf
+
+    Args:
+        model (nn.Module): The model to be attacked.
+        device: The device to run the attack on. If None, the default device will be used.
```

Comment:
The device may not need to be provided manually by the user, `torchattacks` can determine the device based on the tensor input by the user except for a few cases. 🤪

Reply:
Ok, I will remove it from the FMN init 👌